frontend/src/lib/components/sidebar/CriticalAlertModal.svelte

```diff
@@ -0,0 +1,264 @@
+	}
+
+	async function fetchAlerts(pageNumber: number) {
+		isRefreshing = true
+		try {
+			const newAlerts = await SettingService.getCriticalAlerts({
+				page: pageNumber,
+				pageSize: pageSize,
+				acknowledged: hideAcknowledged ? false : undefined
+			})
+
+			alerts = newAlerts
+			hasMore = newAlerts.length === pageSize
+			page = pageNumber
+			updateHasUnaknowledgedCriticalAlerts()
+		} finally {
+			setTimeout(() => {
+				isRefreshing = false
+			}, 500)
+		}
+	}
+
+	async function getAlerts(reset?: boolean) {
+		if (reset) page = 1
+		await fetchAlerts(page)
+	}
+
+	async function checkCriticalAlertChannels() {
+		const channels = (await SettingService.getGlobal({ key: 'critical_error_channels' })) as any[]
+		hasCriticalAlertChannels = channels && channels.length > 0
+	}
+
+	onMount(() => {
+		updateHasUnaknowledgedCriticalAlerts()
+		checkForNewAlertsInterval = setInterval(() => {
+			updateHasUnaknowledgedCriticalAlerts(true)
+		}, 15000)
+	})
+
+	onDestroy(() => {
+		clearInterval(checkForNewAlertsInterval)
+	})
+
+	async function updateHasUnaknowledgedCriticalAlerts(sendToast: boolean = false) {
+		if (checkingForNewAlerts) return
+		checkingForNewAlerts = true
+
+		try {
+			// check if there is at least one unaknowledged alert
+			const unaknowledged = await SettingService.getCriticalAlerts({
+				page: 1,
+				pageSize: 10,
+				acknowledged: false
+			})
+
+			if (numUnaknowledgedCriticalAlerts === 0 && unaknowledged.length > 0 && sendToast) {
+				sendUserToast(
+					'Critical Alert:',
+					true,
+					[
+						{
+							label: 'View',
+							callback: () => {
+								open = true
+							}
+						},
+						{
+							label: 'Acknowledge',
+							callback: () => {
+								if (unaknowledged[0].id) acknowledgeAlert(unaknowledged[0].id)
+							}
+						}
+					],
+					unaknowledged[0].message,
+					10000
+				)
+				if (open && checkingForNewAlerts) getAlerts(false)
+			}
+			numUnaknowledgedCriticalAlerts = unaknowledged.length
+		} finally {
+			checkingForNewAlerts = false
+		}
+	}
+
+	async function acknowledgeAlert(id: number) {
+		await SettingService.acknowledgeCriticalAlert({ id })
+		getAlerts(false)
+	}
+
+	function formatDate(dateString: string | undefined): string {
+		if (!dateString) return ''
+		const date = new Date(dateString)
+		return new Intl.DateTimeFormat('en-US', {
+			year: 'numeric',
+			month: 'short',
+			day: 'numeric',
+			hour: '2-digit',
+			minute: '2-digit',
+			second: '2-digit'
+		}).format(date)
+	}
+
+	async function refreshAlerts() {
+		checkCriticalAlertChannels()
+		await getAlerts(true)
+	}
+
+	function goToPreviousPage() {
+		if (page > 1) {
+			fetchAlerts(page - 1)
+		}
+	}
+
+	function goToNextPage() {
+		if (hasMore) {
+			fetchAlerts(page + 1)
+		}
+	}
+
+	function goToCoreTab() {
+		goto('/#superadmin-settings')
+		instanceSettingsSelectedTab.set('Core')
+	}
+</script>
+
+<Modal bind:open title="Critical Alerts" cancelText="Close" style="max-width: 66%;">
```

Comment:
That's a lot of logic to be mounted all the time, split the logic that needs to decide to open the modal or not versus the logic of the modal itself when opened